python/sidecar/search.py

```diff
@@ -0,0 +1,65 @@
+"""
+Author: Shaurya
+This script performs a search on Semantic Scholar and retrieves information about a set of papers.
+"""
+from semanticscholar import SemanticScholar
+from stopwords import stopwords
+import sys
+import json
+
+class Searcher:
+    # Initialize the SemanticScholar API client and stopwords set
+    def __init__(self):
+        self.s2 = SemanticScholar()
+        self.stopwords = stopwords
+
+    # Method to preprocess the query
+    def preprocess_query(self, query):
+        # convert query to lowercase and remove stopwords
+        query = query.lower()
+        query = "+".join([word for word in query.split() if word not in self.stopwords])
+        return query
+
+'''
+Method to perform the search
+Other search fields can be found here:
+https://github.com/danielnsilva/semanticscholar/blob/da433a1f45e0c17bbd58300de0602c969602675a/semanticscholar/Paper.py#L93
+'''
+def search(
+    query,
+    limit=10,
+    returned_fields=[
+        "title",
+        "abstract",
+        "venue",
+        "year",
+        "paperId",
+        "citationCount",
+        "openAccessPdf",
+        "authors"
+    ],
+):
+    '''
+    This function performs a search on Semantic Scholar's API and 
+    writes the results in JSON to stdout.
+    '''
+
+    # Create a Search object
+    search = Searcher()
+    # Preprocess the query (e.g., lowercase, remove stopwords)
+    query = search.preprocess_query(query)
+    # Perform the search using Semantic Scholar's API
+    results = search.s2.search_paper(query, limit=limit, fields=returned_fields)
```

Comment:
Maybe add some `log` statements here

Reply:
See examples in `ingest.py`

Reply:
added logging - https://github.com/refstudio/refstudio/pull/250/commits/faac256ca624e3846a6d907e64e259863d7afffb